polaris-core/src/main/java/org/apache/polaris/core/PolarisConfiguration.java

```diff
@@ -191,4 +191,12 @@ public static <T> Builder<T> builder() {
               "If set to true, allows tables to be dropped with the purge parameter set to true.")
           .defaultValue(true)
           .build();
+
+  public static final PolarisConfiguration<Boolean> SUPPORT_WASB_CATALOG =
```

Comment:
I think we should re-examine this approach. It's not clear to me how this pattern can extend to having more than 2 schemes -- for example S3 has `s3n`, `s3a`, and `s3`. Also, conceivably, you might want to allow just `abfss` and not `abfs` or vice versa.

Reply:
Agree that the current approach is not at all extensible. I was hoping we have a test annotation like `@WithConfig(configName="true")` that could directly modify the config store, but that would require a larger test framework change that would fall out of scope of this PR. Currently the config store only contains getter methods and there's also no way to modify its content via code via a setter method, so I had to proceed with this pattern of initiating two `DropwizardAppExtension` instances.

Do you think we could wait until the test framework to be more interactable with the config store that we refactor this test?

Reply:
I actually thought that test with the two `DropwizardAppExtension` was pretty good given what we have now. The suggestion you made for an annotation sounds great, but I definitely don't think that's a blocker for this PR.

My comment here is specifically about having a single boolean config specific to WASB (+WASBS) and how that doesn't scale.

Reply:
I don't know that we need to extend this configuration to support `s3n`, etc. I _do_ think we could push the check into the `StorageLocation` class so that it can be responsible for tracking which scheme(s) it supports and which should be rejected at creation. So, e.g., the `AzureLocation` class can implement a `canCreateNew()` method or something and it can, internally, determine whether `wasb` is supported or not. Some future `S3StorageLocation` can track `s3x` URLs, though AFAIK there's no reason to explicitly add/deny support for any of those variations are supported by the existing Iceberg `S3FileIO`.

Reply:
I think we probably should support it. Indeed, don't we already block `http` paths?

Also, why is this specified to catalog creation? The approach here just seems very specific to me.

Reply:
I agree that rationalizing this with the `StorageLocation` or `FileIO` or `StorageConfigurationInfo` or `StorageType` types is a good idea. Right now it seems like the logic for dealing with different schemes and file systems is not centralized enough.